presentation-api/controlling-ua/PresentationConnection_onclose-manual.https.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+
+<meta charset="utf-8">
+<title>Closing a PresentationConnection</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="author" title="He Yue" href="mailto:yue.he@intel.com">
+<link rel="author" title="Tomoyuki Shimizu" href="https://github.com/tomoyukilabs/">
+<link rel="help" href="https://w3c.github.io/presentation-api/#closing-a-presentationconnection">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="common.js"></script>
+<script src="support/stash.js"></script>
+<h2>Description</h2>
+<p>
+  This test validates that after connection close,<br/>
+  the connection state is set closed,<br/>
+  the onclose EventHandler is triggered.
+</p>
+<br/>
+<p>Click the button below to start the test.</p>
+<button id="presentBtn">Start Presentation Test</button>
+
+<script>
+  setup({explicit_timeout: true});
+
+  const presentBtn = document.getElementById('presentBtn');
+
+  promise_test(t => {
+    const clickWatcher = new EventWatcher(t, presentBtn, 'click');
+    const request = new PresentationRequest(presentationUrls);
+    const stash = new Stash(stashIds.toController, stashIds.toReceiver);
+    let connection, eventWatcher;
+
+    t.add_cleanup(() => {
+      if (connection) {
+        connection.onconnect = () => { connection.terminate(); };
+        if (connection.state === 'closed')
+          request.reconnect(connection.id);
+        else
+          connection.terminate();
+      }
+      stash.stop();
+    });
+
+    const checkCloseEvent = evt => {
+      assert_true(evt instanceof PresentationConnectionCloseEvent, 'An event using PresentationConnectionCloseEvent is fired.');
+      assert_true(evt.isTrusted, 'The event is a trusted event.');
+      assert_false(evt.bubbles, 'The event does not bubbles.');
+      assert_false(evt.cancelable, 'The event is not cancelable.');
+      assert_equals(evt.type, 'close', 'The event name is "close".');
+      assert_equals(evt.target, connection, 'event.target is the presentation connection.');
+      assert_equals(connection.state, 'closed', 'State of the presentation connection is "closed".');
+      assert_equals(evt.reason, 'closed', 'The reason for closing the presentation connection is "closed".');
+    };
+
+    return clickWatcher.wait_for('click').then(() => {
+      presentBtn.disabled = true;
+
+      return request.start();
+    }).then(c => {
+      // Enable timeout again, cause no user action is needed from here.
+      t.step_timeout(() => {
+          t.force_timeout();
+          t.done();
+      }, 10000);
+
+      connection = c;
+      return stash.init();
+    }).then(() => {
+      // Step 1: close the presentation connection in "connecting" state
```

Comment:
Oops, that's right. I'll move `stash.init()` somewhere else.

By the way, while I'm not sure whether it is correct or not, Chrome Canary throws an exception like `"InvalidAccessError: PresentationRequest::start() requires user gesture."` when `request.start()` is called within `stash.init().then()`. This is the reason why I avoided calling `stash.init()` before `request.start()`.